### DISCUSSION TITLE
Removing Get prefixed methods in favor of Properties

### DISCUSSION THREAD

**Discussion by geek911** [dotnet/roslyn#50134] [Ideas]
One of the most confusing part to new .Net developers is when to Prefix a Get to a method instead of using a property. In my opinion it makes more sense to refectore the CoreFX to use Properties instead of Get methods. For example, `GetHashCode()` can be replace with a property `HashCode`. What do you think? Is it a good idea?

**Comment by huoyaoyuan** [comment]
First, existing apis can never change.

Although property getters and GetXXX methods are close in the signature, there are different convensional expectation to them.
Properties are expected to be **quick, reproducible and no side-effect**. So the debugger can show a property at any time without worrying about those. Methods are expected to have calculations to do, or tracking the newly created object.
